src/filters/factory.rs

```diff
@@ -16,12 +16,22 @@
 
 use prometheus::Registry;
 
-use crate::filters::{ConfigType, Error, Filter};
+use crate::{
+    config::ConfigType,
+    filters::{Error, Filter},
+};
 
 /// An owned pointer to a dynamic [`FilterFactory`] instance.
 pub type DynFilterFactory = Box<dyn FilterFactory>;
 
 /// Provides the name and creation function for a given [`Filter`].
+///
+/// An implementation of [`FilterFactory`] provides a `name` and
+/// `create_filter` method.
+///
+/// - The `create_filter` method takes in [configuration][filter configuration]
+///   for the filter to create and returns a new instance of its filter type.
+/// - The `name` method returns the unique identifier of the filter.
```

Comment:
```suggestion
```
This seems redundant to list and describe the trait methods here and repeat them alongside the functions themselves